CvGameCoreDLL_Expansion2/CvAStar.cpp

```diff
@@ -2198,62 +2198,38 @@ static int BuildRouteVillageBonus(CvPlayer* pPlayer, CvPlot* pPlot, RouteTypes e
 	if (pPlot->getResourceType() != NO_RESOURCE)
 		return 0;
 
-	int iBonus = 0;
+	// No villages on mountains
+	if (pPlot->getTerrainType() == TERRAIN_MOUNTAIN)
+		return 0;
+
+	// No villages for China near cities, no villages for Brazil on jungle/forest, no villages for Netherlands on marshes.
+	if (!eBuilderTaskingAi->MayWantVillageOnPlot(pPlot))
+		return 0;
 
+	// If we have a town or village here, give a big bonus
 	ImprovementTypes eImprovement = pPlot->getImprovementType();
 	if (eImprovement != NO_IMPROVEMENT)
 	{
 		CvImprovementEntry* pkImprovementInfo = GC.getImprovementInfo(eImprovement);
+
 		if (pkImprovementInfo != NULL)
 		{
 			for (int iI = 0; iI < NUM_YIELD_TYPES; iI++)
```

Comment:
this still seems expensive - typically it's enough to iterate up to YIELD_TOURISM

Reply:
Not YIELD_GOLDEN_AGE_POINTS?

Reply:
Can change, really it was the canBuild call that was causing issues. Changing the loop cutoff might shave off a millisecond or two per turn I suppose.